conans/test/utils/tools.py

```diff
@@ -659,14 +661,37 @@ def scm_info_cache(self, reference):
 
         if not isinstance(reference, ConanFileReference):
             reference = ConanFileReference.loads(reference)
-        layout = self.cache.package_layout(reference)
+        layout = self.get_latest_ref_layout(reference)
         content = load(layout.conandata())
         data = yaml.safe_load(content)
         if ".conan" in data:
             return self._create_scm_info(data[".conan"])
         else:
             return self._create_scm_info(dict())
 
+    def get_latest_prev(self, ref: ConanReference) -> PackageReference:
+        """Get the latest PackageReference given a ConanReference"""
+        latest_rrev = self.cache.get_latest_rrev(ref)
+        all_package_ids = self.cache.get_package_ids(latest_rrev)
+        return self.cache.get_latest_prev(all_package_ids[0])
+
+    def get_latest_pkg_layout(self, pref: PackageReference) -> PackageLayout:
+        """Get the latest PackageLayout given a file reference"""
+        # Let's make it easier for all the test clients
+        if pref.ref.revision is None:
+            pref_ = PackageReference(self.cache.get_latest_rrev(pref.ref), pref.id)
+        else:
+            pref_ = pref
```

Comment:
@memsharded @czoido here, I had doubts about assuming that we're always passing the pref with the recipe revision or helping test clients to pass it or not, what do you think?

Reply:
I think that I would opt for assuming that the PackageReference comes with the rrev and fail if it doesn't. Then, if you have to get the latest rrev it could be done in the test before calling this method.

Reply:
Agree but the thing is that if I request a pref with recipe revision included, almost all the occurrences will look like:

```python
pref = client.get_latest_pref(ANY_REF)
pkg_layout = client.get_pkg_layout(pref)
```
So, totally agree with these two lines will appear a lot of times, right?

Reply:
you say ``get_latest_pref()`` Is that a typo, you mean ``get_latest_prev()``?

The main problem in the above is not the latest revision, but the ``all_package_ids[0]``. It doesn't make sense to pick the first (and assuming only) package-id in the cache, so the package_id needs to be explicit (it is relatively easy to capture it in the output while creating the package). The latest revisions, both recipe and package could be automated in the ``TestClient``, IMO.

Reply:
Ok, I'm passing the tests locally and it's not happening for all the cases so it should not be a big deal. Applying the changes...